File: src/main/java/com/lessmarkup/dataobjects/Smile.java
package com.lessmarkup.dataobjects;

import com.lessmarkup.interfaces.data.AbstractDataObject;
import com.lessmarkup.interfaces.recordmodel.RequiredField;

public class Smile extends AbstractDataObject {
    @RequiredField
    private String contentType;
    @RequiredField
    private byte[] data;
    @RequiredField
    private String name;
    @RequiredField
    private String code;

    public String getContentType() {
        return contentType;
    }

    public void setContentType(String contentType) {
        this.contentType = contentType;
    }

    public byte[] getData() {
        return data;
    }

    public void setData(byte[] data) {
        this.data = data;
    }

    public String getName() {
        return name;
    }

    public void setName(String name) {
        this.name = name;
    }

    public String getCode() {
        return code;
    }

    public void setCode(String code) {
        this.code = code;
    }
}

File: src/main/scala/com/lessmarkup/dataobjects/Smile.scala
/*
 * This Source Code Form is subject to the terms of the Mozilla Public License, v. 2.0.
 * If a copy of the MPL was not distributed with this file, You can obtain one at
 * http://mozilla.org/MPL/2.0/.
 */

package com.lessmarkup.dataobjects

import com.lessmarkup.interfaces.annotations.RequiredField
import com.lessmarkup.interfaces.data.{BinaryData, DataObject}

class Smile extends DataObject {
  @RequiredField
  var contentType: String = null
  @RequiredField
  var data: BinaryData = null
  @RequiredField
  var name: String = null
  @RequiredField
  var code: String = null
}